src/main/kotlin/com/avail/interpreter/levelTwo/operation/L2_STRIP_MANIFEST.kt

```diff
@@ -0,0 +1,109 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.avail.interpreter.levelTwo.operation
+
+import com.avail.interpreter.levelTwo.L2Instruction
+import com.avail.interpreter.levelTwo.L2NamedOperandType
+import com.avail.interpreter.levelTwo.L2OperandType
+import com.avail.interpreter.levelTwo.L2Operation
+import com.avail.interpreter.levelTwo.operand.L2ReadBoxedVectorOperand
+import com.avail.interpreter.levelTwo.register.L2Register
+import com.avail.optimizer.L2ValueManifest
+import com.avail.optimizer.jvm.JVMTranslator
+import com.avail.optimizer.values.L2SemanticValue
+import org.objectweb.asm.MethodVisitor
+import java.util.*
+
+/**
+ * This is a helper operation which produces no JVM code, but is useful to limit
+ * which [L2Register]s and [L2SemanticValue]s are live when reaching
+ * a back-edge.
+ *
+ * @author Mark van Gulik &lt;mark@availlang.org&gt;
+ */
+object L2_STRIP_MANIFEST : L2Operation(
+	L2OperandType.READ_BOXED_VECTOR.`is`("live values"))
+{
+	// Prevent this instruction from being removed, because it constrains
+	// the manifest along a back-edge, even after optimization.
+	override fun hasSideEffect(): Boolean = true
+
+	override fun instructionWasAdded(
+		instruction: L2Instruction,
+		manifest: L2ValueManifest)
+	{
+		val liveVector = instruction.operand<L2ReadBoxedVectorOperand>(0)
+
+		// Clear the manifest, other than the mentioned semantic values and
+		// registers.
+		val liveSemanticValues: MutableSet<L2SemanticValue> = HashSet()
+		val liveRegisters: MutableSet<L2Register> = HashSet()
+		for (read in liveVector.elements())
+		{
+			liveSemanticValues.add(read.semanticValue())
+			liveRegisters.add(read.register())
+		}
+		manifest.retainSemanticValues(liveSemanticValues)
+		manifest.retainRegisters(liveRegisters)
+		liveVector.instructionWasAdded(manifest)
+	}
+
+	override fun updateManifest(
+		instruction: L2Instruction,
+		manifest: L2ValueManifest,
+		optionalPurpose: L2NamedOperandType.Purpose?)
+	{
+		assert(this == instruction.operation())
+		assert(optionalPurpose == null)
+		val liveVector = instruction.operand<L2ReadBoxedVectorOperand>(0)
+
+		// Clear the manifest, other than the mentioned semantic values and
+		// registers.
+		val liveSemanticValues: MutableSet<L2SemanticValue> = HashSet()
+		val liveRegisters: MutableSet<L2Register> = HashSet()
+		for (read in liveVector.elements())
```

Comment:
Something like this:
```
val liveSemanticValues = liveVector.elements().mapTo(mutableSetOf()) { it.semanticValue() }
val liveRegisters = liveVector.elements().mapTo(mutableSetOf()) { it.register() }
```

Reply:
Ok, I updated these. I know I didn't catch all items of this sort; hopefully we'll clean up the ones I missed via the review